src/http/Response.cpp

```diff
@@ -0,0 +1,208 @@
+#include "Response.hpp"
+
+namespace http
+{
+
+Response::Response(void) {}
+
+Response::~Response(void) {}
+
+void
+Response::load_http_request(Request &req)
+{
+	init_response_map();
+	std::string requested_path = req.get_path();
+	_response_map["dir_location"] += requested_path;
+	std::cout << "dir_location at load_HTTP_REQUEST : " << _response_map["dir_location"] << std::endl;
+	std::cout << "METHOD: " << req.get_method() << "\nAuth: " << req.method_is_authorized(req.get_method())
+			  << std::endl;
+	if (!req.method_is_authorized(req.get_method()))
+	{
+		load_response_map(405);
+	}
+	else if (!file_exists(_response_map["dir_location"]))
+	{
+		load_response_map(404);
+	}
+	else
+	{
+		load_response_map(200);
+	}
+}
+
+void
+Response::init_response_map(void)
+{
+	_response_map["Status-line"] = "";
+	_response_map["Date"] = "";
+	_response_map["Server"] = "Webserver";
+	_response_map["Content-Length"] = "";
+	_response_map["Content-Type"] = "";
+	_response_map["Connection"] = "Closed";
+	_response_map["Protocol"] = "HTTP/1.1 ";
+	_response_map["header-string"] = "";
+	_response_map["body-string"] = "";
+	_response_map["full-response-string"] = "";
+	_response_map["dir_location"] = "/Users/rburri/Desktop/network_cpp";
+}
+
+void
+Response::load_response_map(int status_code)
+{
+	_response_map["Date"] += get_time_stamp();
+	_response_map["Status-line"]
+		= _response_map["Protocol"] + _status_code.get_key_value_formated(status_code);
+	if (status_code != 200)
+	{
+		set_response_type(".html");
+		create_error_html_page(status_code);
+	}
+	else
+	{
+		set_response_type(_response_map["dir_location"]);
+		construct_body_string(_response_map["dir_location"]);
+	}
+	set_content_length(_response_map["body-string"]);
+	construct_header_string();
+	construct_full_response();
+}
+
+std::string
+Response::get_time_stamp(void)
+{
+	std::time_t stamp = std::time(NULL);
+	std::string formated_date = std::asctime(std::gmtime(&stamp));
+	formated_date = formated_date.substr(0, formated_date.length() - 1);
+	formated_date += " GMT";
+	return formated_date;
+}
+
+void
+Response::set_content_length(std::string str)
+{
+	_response_map["Content-Length"] = std98::to_string(str.length());
+}
+
+bool
+Response::file_exists(std::string path)
+{
+	std::ifstream file;
+ 	bool		  ret = false;
+	file.open(path.c_str());
+	if (file)
```

Comment:
cppreference.cpp: [`is_open()`](https://en.cppreference.com/w/cpp/io/basic_fstream/is_open) ?
```suggestion
	if (file.is_open())
```

Reply:
makes sense.